tokio/src/io/ready.rs

```diff
@@ -240,11 +282,25 @@ impl ops::Sub<Ready> for Ready {
 
 impl fmt::Debug for Ready {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-        fmt.debug_struct("Ready")
-            .field("is_readable", &self.is_readable())
-            .field("is_writable", &self.is_writable())
-            .field("is_read_closed", &self.is_read_closed())
-            .field("is_write_closed", &self.is_write_closed())
-            .finish()
+        #[cfg(any(target_os = "linux", target_os = "android"))]
```

Comment:
You shouldn't need to dup the full block. You can start the `fmt` with:

```rust
let fmt = fmt.debug_struct("Ready");`
```

then have all the fields that are common across platforms.

Then:

```rust
#[cfg(any(target_os = "linux", target_os = "android"))]
fmt.field("is_priority", &self.is_priority_ready())
```

Reply:
ah nice. I tried this before in a slightly different way and ran into lifetime issues.